File: LeetCode/cpp/804.cpp
//Link: https://leetcode.com/problems/unique-morse-code-words/

class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        string morse[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",
                          ".---","-.-",".-..","--","-.","---",".--.","--.-",
                          ".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        set<string> morse_words;
        string transformation;
        for(int i = 0 ; i < words.size() ; i++) {
            transformation = "";
            for(int j = 0 ; j < words[i].length() ; j++)
                transformation += morse[words[i][j] - 'a'];
            morse_words.insert(transformation);
        }
        return morse_words.size();
    }
};

File: LeetCode/py/804.py
#Link: https://leetcode.com/problems/unique-morse-code-words/

class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---",\
                 "-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-",\
                 "..-","...-",".--","-..-","-.--","--.."]
        morse_word = set()
        for word in words:
            transformation = ""
            for ch in word:
                transformation += morse[ord(ch) - ord('a')]
            morse_word.add(transformation)
        return len(morse_word)
        